build/android/config.gni

```diff
@@ -135,3 +135,17 @@ brave_jni_headers_sources = [
   "//brave/browser/brave_ads/android/java/org/chromium/chrome/browser/BraveAdsNativeHelper.java",
   "//brave/components/safetynet/java/org/chromium/chrome/browser/util/SafetyNetCheck.java",
 ]
+
+# Additional deps for //components/search_engines/android:java to make these tests succeed on cr130:
+#   PageMetricsUnitTest.DomainsLoadedCount
+#   PageMetricsUnitTest.BookmarkCount
+#   PageMetricsUnitTest.FirstPageLoadTimeTooLate
+#   PageMetricsUnitTest.FirstPageLoadTimeImmediate
+#   PageMetricsUnitTest.PagesLoadedCount
+#   PageMetricsUnitTest.FirstPageLoadTimeLater
+#   BraveSearchProviderTest.SearchDoesNotIncludeHistoryWhenHistoryDisabled
+#   BraveSearchProviderTest.SearchIncludesHistoryWhenHistoryEnabled
+# Related Chromium change: https://github.com/chromium/chromium/commit/17b1d05d2ccda19c3ebd903075227bc8e851acf0
+# Tests were crashing because they cannot create SearchEngineCountryDelegateImpl object
+brave_components_search_engines_android_java_deps =
```

Comment:
I'm confused, is this an upstream issue?

Reply:
cc: @AlexeyBarabash

Reply:
I see that it's needed for `BraveSearchProviderTest`, but also needed for `PageMetricsUnitTest` which is an upstream test

Reply:
@bridiver
`PageMetricsUnitTest` is ours -  https://github.com/brave/brave-core/blob/cr130/browser/misc_metrics/page_metrics_unittest.cc 

The original reason of failure is the [line](https://source.chromium.org/chromium/chromium/src/+/main:components/search_engines/android/BUILD.gn;l=22;drc=dabf53d870a39f6ea7b4e5af6368d54a7e8974eb;bpv=0;bpt=0)
`jar_excluded_patterns = [ "*/SearchEngineCountryDelegateImpl.class" ]`

The non-test application runs well because the target `"//components/search_engines/android:delegate_public_impl_java"` specified as deps for `//chrome/android:delegate_public_impl_java`, [link](https://source.chromium.org/chromium/chromium/src/+/main:chrome/android/BUILD.gn;l=315;drc=dabf53d870a39f6ea7b4e5af6368d54a7e8974eb;bpv=0;bpt=0?q=chrome%2Fandroid%2FBUILD.gn&ss=chromium%2Fchromium%2Fsrc).

Any my attemps to add `"//components/search_engines/android:delegate_public_impl_java"` as deps into other places like `//brave/test/brave_public_test_apk` to avoid patch - did not fix the failed tests.

Reply:
why are these tests using the real factories for testing? That seems to be the root of the issue here https://github.com/brave/brave-core/blob/cr130/browser/misc_metrics/page_metrics_unittest.cc#L49C25-L49C57
because `SearchEngineChoiceService` depends on `SearchEngineCountryDelegateImpl`.

Reply:
this is definitely not the right fix here because they have intentionally excluded `SearchEngineCountryDelegateImpl` from this target and now we're adding it back in.

Reply:
> Did you try adding it as a dep for brave_unit_tests similar to how it's added for brave_browser_tests?

No, I didn't. When I just tried at 
https://github.com/brave/brave-core/pull/25340/commits/72d64e2543aeb08d6a1d2e584a7e05bca064b0b9